Add Rails Docker Compose example

CodySanders opened the pull request.

CodySanders committed:
Add Rails Docker Compose example

townofdon commented on docker-compose-rails/.env-vars:
```diff
@@ -0,0 +1 @@
+POSTGRES_PASSWORD="really-good-password"
```
the best

townofdon commented:
@CodySanders This is awesome! 👏 

Kind of a nit thing - for naming consistency could you change `docker-compose-rails` to `example-rails`?

townofdon commented on README.md:
```diff
@@ -8,7 +8,6 @@
 - Add node.js Express / PostgreSQL example
 - Add React (CRA) example
 - Add React (Next.js) example
-- Add Rails example
```
Could you add a link to the rails example to line 4 of the readme? Also could we move the rails example to `./example-rails/`?

townofdon commented on docker-compose-rails/Dockerfile:
```diff
@@ -0,0 +1,18 @@
+FROM ruby:3.0.1
+
+RUN apt update -qq && apt install -y nodejs npm postgresql-client
+RUN npm install --global yarn
```
Since this is meant to be a tutorial of sorts, would you be up for adding some comments for what the various lines do? I went this direction in the [php example](https://github.com/teamairship/docker-examples/blob/main/example-php-apache/Dockerfile) - what do you think? I know it's verbose but for someone totally new to Docker, it would be helpful to have a line-by-line explanation.

townofdon commented on docker-compose-rails/docker-compose.yml:
```diff
@@ -0,0 +1,18 @@
+services:
+  db:
```
Would you be up for adding line-by-line comments/instructions to the docker compose file? I went with this approach [here](https://github.com/teamairship/docker-examples/blob/main/example-php-apache/docker-compose.yml) and it seemed to be good for learning purposes. Let me know what you think!

townofdon reviewed (changes requested):
This is great! I had a few suggestions/comments for learning purposes to make this easier to digest for someone completely new to Docker. Thanks for adding this!

CodySanders committed:
Rename project root for naming consistency

CodySanders committed:
Add public assets/packs so a local compile is not needing before running docker compose

CodySanders committed:
Add link to Rails example dir in README

CodySanders committed:
Add comments in Dockerfile and docker-compose.yml to give more context

CodySanders commented:
@townofdon I think I addressed all of the suggestions but let me know if I missed anything. I think the comments are certainly helpful but it would also be my hope that this project would be a jumping off point for a discussion with someone who is looking to get into Docker and using it in their development process. There are still many small nuances that aren't covered in detail here but I think it could be a good start.

CodySanders commented:
Also, if you pull this down and try to run it please let me know if you run into any issues. I think I remembered to include everything needed but there is always the chance that my machine already had a forgotten dependency included that a new machine would not

townofdon commented:
@CodySanders Awesome, thanks for adding those comments! I agree that the documentation is needed for understanding the nuances and anyone looking to deep-dive into Docker, but the comments definitely help and especially add context.

I checked out the branch and ran `docker-compose up` - worked like a charm!

However running the db:create and db:migrate scripts resulted in an err:

![image](https://user-images.githubusercontent.com/5880655/118505613-930b4280-b6fa-11eb-8f71-451debfe951c.png)

I saw that POSTGRES_PASSWORD is in .env-vars, maybe another env var is needed?

CodySanders commented:
Hmm, I will take a look at that. If the `.env-vars` file is being included in those commands correctly (which I believe it should be since it is declared for those services in `docker-compose.yml`) then it should just work out of the box. I'll let you know what I find out.
